File: Python Snippets/bubble_sort.py
from random import randint

def bubble_sort(nums):
  swapped = True

  while(swapped):
    swapped = False
    for i in range(len(nums) - 1):
      if nums[i] > nums[i + 1]:
        nums[i], nums[i + 1] = nums[i + 1], nums[i]
        swapped = True

  return nums

sample_array = [randint(-10, 10) for i in range(10)]
print(f"Unsorted array: {sample_array}")

sorted_array = bubble_sort(sample_array)
print(f"Sorted array: {sorted_array}")

File: R Snippets/bubble_sort.r
bubble_sort <- function(nums) {
  swapped <- TRUE

  while (swapped) {
    swapped <- FALSE

    for (index in 1:(length(nums)-1)) {
      if (nums[index] > nums[index + 1]) {
        temp <- nums[index]
        nums[index] <- nums[index + 1]
        nums[index + 1] <- temp
        swapped <- TRUE
      }
    }
  }

  return (nums)
}

sample_array <- sample(0:50, 25, replace=TRUE)
cat("Unsorted Array:", sample_array, "\n")

sorted_array <- bubble_sort(sample_array)
cat("Sorted Array:  ", sorted_array, "\n")
